src/ResourceManager/Compute/Commands.Compute/Strategies/AsyncCmdletExtensions.cs

```diff
@@ -79,20 +81,40 @@ public static void WriteVerbose(this IAsyncCmdlet cmdlet, string message, params
                 });
         }
 
-        sealed class AsyncCmdlet : IAsyncCmdlet
+        sealed class AsyncCmdlet<T> : IAsyncCmdlet
+            where T : PSCmdlet
         {
             public SyncTaskScheduler Scheduler { get; } = new SyncTaskScheduler();
 
-            readonly Cmdlet _Cmdlet;
+            readonly T _Cmdlet;
 
             public List<ITaskProgress> TaskProgressList { get; } 
                 = new List<ITaskProgress>();
 
-            public AsyncCmdlet(Cmdlet cmdlet)
+            public AsyncCmdlet(T cmdlet)
             {
                 _Cmdlet = cmdlet;
             }
 
+            public bool WhatIf
```

Comment:
@sergey-shandar As we have discussed, this is an anti-pattern, we should not print additional information based on our own setting of WhatIf - WhatIf should be handled automaticsally by the PowerShell runtime.  Additional information printed in WhatIf should use only ShouldProcess prompts.

Verbose output should simply be written to the verbose stream - we cna log such output to the Verbose or Debug streams, depending on the level of information (verbose versus debug/very verbose)